src/analysis/sum_stat.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import os
+
+import pandas
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import subprocess
+import re
+import matplotlib.pyplot as plt
+
+logging.basicConfig(level=logging.INFO, format='%(levelname)s:%(asctime)s %(message)s')
+
+
+class SummaryStat(object):
+    '''
+    A class to easily aggregate multiple summary statistics files.
+    '''
+
+    def __search(self, regex: str, l: list, pvalue=False):
+        results_cords = []
+        results_names = []
+        for i, name in enumerate(l):
+            if re.search(regex, name):
+                if pvalue:
+                    results_cords.append(i + 1)
+                    results_names.append(name)
+                else:
+                    return i + 1, name
+        if pvalue:
+            return results_cords, results_names
+        return False, False
+
+    def __get_ss_info(self, header: list, tag: str):
+        pos_i, pos = self.__search("^[Pp][Oo][Ss]*", header)
+        id_i, id = self.__search("^[Ii][Dd]*", header)
+        p_i, p = self.__search("^[Pp].*val*|^[Ll][Oo][Gg].*[Pp]*|^[Pp]$", header, pvalue=True)
+
+        ss_info = {'pos': pos, 'id': id, 'p': p,
+                   'pos_i': pos_i, 'id_i': id_i, 'p_i': p_i,
+                   'tag': tag}
+        cols = [pos, id]
+        for item in cols:
+            if not item:
+                cols.remove(item)
+        for pval in p:
+            if not p:
+                break
+            cols.append(pval)
+
+        ss_info['cols'] = cols
+        return ss_info
+
+    def __get_dataframe(self, ss: str, ss_info: dict):
+        df = pd.read_table(ss + '.filtered', skipinitialspace=True, usecols=ss_info[ss]['cols'])
+        df_filtered = \
+            df.drop_duplicates(subset=[ss_info[ss]['pos']]) \
+                .sort_values(by=[ss_info[ss]['pos']]) \
+                .reset_index(drop=True)
+
+        for pvalue in ss_info[ss]['p']:
+            if df_filtered[pvalue].max() > 1:
+                df_filtered[pvalue] = np.log10(df_filtered[pvalue])
+            if df_filtered[pvalue].min() < 0:
+                df_filtered[pvalue] = df_filtered[pvalue].apply(lambda x: x * -1)
+
+        keep_same = {ss_info[ss]['pos'], ss_info[ss]['id']}
+        df_filtered.columns = ['{}{}'.format(c, '' if c in keep_same else '_'+ss_info[ss]['tag']) for c in df_filtered.columns]
+        return df_filtered
+
+    def __merge_data(self, df_dict: dict, ss_info: dict):
```

Comment:
please add docs

Reply:
I don't think providing docs for private methods is necessary, because they won't be directly used and i made them only to increase main function (`prepare_data`) readability.